File: sdk/go/examples/mint_user_token.go
// Example: Minting Stigmer user tokens with PlatformClient credentials.
//
// This example demonstrates how a platform builder's Go backend mints
// Stigmer-signed JWTs for their users. The tokens are returned to the
// frontend and passed to the React SDK's StigmerProvider via getAccessToken.
//
// This is NOT a runnable program — it shows the API patterns.
package examples

import (
	"context"
	"fmt"
	"log"
	"os"

	stigmer "github.com/stigmer/stigmer/sdk/go"
)

func MintUserToken() {
	ctx := context.Background()

	auth, err := stigmer.NewPlatformClientAuth(
		stigmer.WithPlatformClientCredentials(
			os.Getenv("STIGMER_CLIENT_ID"),
			os.Getenv("STIGMER_CLIENT_SECRET"),
		),
	)
	if err != nil {
		log.Fatal(err)
	}
	defer auth.Close()

	result, err := auth.MintUserToken(ctx, &stigmer.MintUserTokenInput{
		UserID:    "user-123",
		UserEmail: "jane@acme.com",
		UserName:  "Jane Doe",
	})
	if err != nil {
		if stigmer.IsUnauthenticated(err) {
			log.Fatal("Invalid PlatformClient credentials — verify STIGMER_CLIENT_ID and STIGMER_CLIENT_SECRET")
		}
		if stigmer.IsNotFound(err) {
			log.Fatal("User does not have a Stigmer identity account and auto-provisioning is disabled")
		}
		log.Fatal(err)
	}

	fmt.Printf("Token: %s\n", result.AccessToken)
	fmt.Printf("Expires at: %s\n", result.ExpiresAt)
}

File: sdk/typescript/examples/mint-user-token.ts
/**
 * Example: Mint Stigmer user tokens with PlatformClient credentials.
 *
 * This minimal Express server exposes a /api/stigmer-token endpoint
 * that platform builders call from their React frontend. The React SDK's
 * StigmerProvider uses getAccessToken to fetch from this endpoint.
 *
 * Prerequisites:
 *   STIGMER_CLIENT_ID   — PlatformClient client_id
 *   STIGMER_CLIENT_SECRET — PlatformClient client_secret
 *
 * This is NOT a runnable program — it shows the API patterns.
 * A real implementation would add your own auth middleware.
 */

import express from "express";
import { createPlatformClientAuth } from "@stigmer/sdk/node";

const app = express();

const auth = createPlatformClientAuth({
  baseUrl: "https://api.stigmer.ai",
  clientId: process.env.STIGMER_CLIENT_ID!,
  clientSecret: process.env.STIGMER_CLIENT_SECRET!,
});

app.get("/api/stigmer-token", async (req, res) => {
  // In production, authenticate the request with your own auth system
  // and extract the user identity from the session.
  const userId = req.query.userId as string;

  try {
    const { accessToken, expiresAt } = await auth.mintUserToken({
      userId,
      userEmail: req.query.email as string,
      userName: req.query.name as string,
    });

    res.json({ accessToken, expiresAt: expiresAt.toISOString() });
  } catch (error) {
    console.error("Failed to mint token:", error);
    res.status(500).json({ error: "Failed to mint Stigmer token" });
  }
});

app.listen(3001, () => {
  console.log("Token endpoint running at http://localhost:3001/api/stigmer-token");
});
